File: sdks/java/src/main/java/com/vaultagent/sdk/exception/VaultAgentException.java
package com.vaultagent.sdk.exception;

/**
 * Base exception class for Vault Agent SDK
 */
public class VaultAgentException extends Exception {
    
    private final String requestId;
    
    public VaultAgentException(String message) {
        super(message);
        this.requestId = null;
    }
    
    public VaultAgentException(String message, String requestId) {
        super(message);
        this.requestId = requestId;
    }
    
    public VaultAgentException(String message, Throwable cause) {
        super(message, cause);
        this.requestId = null;
    }
    
    public VaultAgentException(String message, Throwable cause, String requestId) {
        super(message, cause);
        this.requestId = requestId;
    }
    
    public String getRequestId() {
        return requestId;
    }
    
    @Override
    public String toString() {
        String result = super.toString();
        if (requestId != null) {
            result += " (Request ID: " + requestId + ")";
        }
        return result;
    }
}

/**
 * Exception thrown when authentication fails
 */
class AuthenticationError extends VaultAgentException {
    public AuthenticationError(String message) {
        super(message);
    }
    
    public AuthenticationError(String message, String requestId) {
        super(message, requestId);
    }
}

/**
 * Exception thrown when authorization fails
 */
class AuthorizationError extends VaultAgentException {
    public AuthorizationError(String message) {
        super(message);
    }
    
    public AuthorizationError(String message, String requestId) {
        super(message, requestId);
    }
}

/**
 * Exception thrown when a resource is not found
 */
class NotFoundError extends VaultAgentException {
    public NotFoundError(String message) {
        super(message);
    }
    
    public NotFoundError(String message, String requestId) {
        super(message, requestId);
    }
}

/**
 * Exception thrown when validation fails
 */
class ValidationError extends VaultAgentException {
    public ValidationError(String message) {
        super(message);
    }
    
    public ValidationError(String message, String requestId) {
        super(message, requestId);
    }
}

/**
 * Exception thrown when rate limit is exceeded
 */
class RateLimitError extends VaultAgentException {
    public RateLimitError(String message) {
        super(message);
    }
    
    public RateLimitError(String message, String requestId) {
        super(message, requestId);
    }
}

/**
 * Exception thrown when connection fails
 */
class ConnectionError extends VaultAgentException {
    public ConnectionError(String message) {
        super(message);
    }
    
    public ConnectionError(String message, Throwable cause) {
        super(message, cause);
    }
    
    public ConnectionError(String message, String requestId) {
        super(message, requestId);
    }
}

File: sdks/dotnet/Exceptions/VaultAgentException.cs
using System;

namespace VaultAgent.SDK.Exceptions
{
    /// <summary>
    /// Base exception class for Vault Agent SDK
    /// </summary>
    public class VaultAgentException : Exception
    {
        public string RequestId { get; }

        public VaultAgentException(string message) : base(message)
        {
        }

        public VaultAgentException(string message, string requestId) : base(message)
        {
            RequestId = requestId;
        }

        public VaultAgentException(string message, Exception innerException) : base(message, innerException)
        {
        }

        public VaultAgentException(string message, Exception innerException, string requestId) : base(message, innerException)
        {
            RequestId = requestId;
        }

        public override string ToString()
        {
            var result = base.ToString();
            if (!string.IsNullOrEmpty(RequestId))
            {
                result += $" (Request ID: {RequestId})";
            }
            return result;
        }
    }

    /// <summary>
    /// Exception thrown when authentication fails
    /// </summary>
    public class AuthenticationException : VaultAgentException
    {
        public AuthenticationException(string message) : base(message) { }
        public AuthenticationException(string message, string requestId) : base(message, requestId) { }
    }

    /// <summary>
    /// Exception thrown when authorization fails
    /// </summary>
    public class AuthorizationException : VaultAgentException
    {
        public AuthorizationException(string message) : base(message) { }
        public AuthorizationException(string message, string requestId) : base(message, requestId) { }
    }

    /// <summary>
    /// Exception thrown when a resource is not found
    /// </summary>
    public class NotFoundException : VaultAgentException
    {
        public NotFoundException(string message) : base(message) { }
        public NotFoundException(string message, string requestId) : base(message, requestId) { }
    }

    /// <summary>
    /// Exception thrown when validation fails
    /// </summary>
    public class ValidationException : VaultAgentException
    {
        public ValidationException(string message) : base(message) { }
        public ValidationException(string message, string requestId) : base(message, requestId) { }
    }

    /// <summary>
    /// Exception thrown when rate limit is exceeded
    /// </summary>
    public class RateLimitException : VaultAgentException
    {
        public RateLimitException(string message) : base(message) { }
        public RateLimitException(string message, string requestId) : base(message, requestId) { }
    }

    /// <summary>
    /// Exception thrown when connection fails
    /// </summary>
    public class ConnectionException : VaultAgentException
    {
        public ConnectionException(string message) : base(message) { }
        public ConnectionException(string message, Exception innerException) : base(message, innerException) { }
        public ConnectionException(string message, string requestId) : base(message, requestId) { }
    }
}